scripts/legacy_expression_tool.py

```diff
@@ -0,0 +1,394 @@
+# 
+#  Author by: Miguel Angel Pedraza
+# 
+
+from pymongo import MongoClient
+from bson import json_util, ObjectId
+import json
+from operator import itemgetter
+import re
+
+from datetime import datetime
+
+import argparse
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def parse_json(data):
+    return json.loads(json_util.dumps(data))
+
+object_replaced_list = []
+find_object_occurrences = []
+found_legacy_expressions = []
+translation_legacy_expressions = []
+object_occurrences_backup = []
+
+# Init time
+now = datetime.now()
+init_time = now.strftime("%y-%m-%d %H:%M:%S")
+
+# Create the CLI argunments parser
+parser = argparse.ArgumentParser(description='Tool to migrate legacy expressions in IoT Agents')
+parser.add_argument('-db','--database', help='Database name', required=True)
+parser.add_argument('-c','--collection', help='Collection name', required=True)
+parser.add_argument('-H','--host', help='Mongodb host', required=False)
+parser.add_argument('-p','--port', help='Mongodb port', required=False)
+parser.add_argument('-t','--translation', help='Translation file', required=False)
+parser.add_argument('-D','--debug', help='Debug mode', required=False, action='store_true')
+parser.add_argument('-cm','--commit', help='Commit changes to database', required=False, action='store_true')
+parser.add_argument('-el','--expressionlanguage', help='How to handle expressionLanguage values. Can be: delete, ignore or jexl', required=False)
+parser.add_argument('-stat','--statistics', help='Show statistics at the end of the execution. Possible values: service subservice', required=False)
+parser.add_argument('-S','--service', help='FIWARE service filter', required=False)
+parser.add_argument('-P','--servicepath', help='FIWARE servicepath filter', required=False)
+parser.add_argument('-i','--deviceid', help='Device ID filter', required=False)
+parser.add_argument('-r','--restore', help='Restore from backup file', required=False)
+args = vars(parser.parse_args())
+
+# Process args
+host = 'localhost'
+if args['host']:
+    host = args['host']
+
+port = '27017'
+if args['port']:
+    port = args['port']
+
+debug = False
+if args['debug']:
+    debug = True
+
+commit = False
+if args['commit'] == True:
+    print('INFO: Running the script in commit mode, this will update the database')
+    commit = True
+
+fiware_service = {'$regex': '.*'}
+if args['service']:
+    fiware_service = args['service']
+    if debug==True:
+        print('Filtering by service: ' + fiware_service)
+
+fiware_servicepath = {'$regex': '.*'}
+if args['servicepath']:
+    fiware_servicepath = args['servicepath']
+    if debug==True:
+        print('Filtering by servicepath: ' + fiware_servicepath)
+
+filter_device_id = ''
+if args['filterdevices']:
+    filter_device_id = args['filterdevices']
+    print('Filtering by device ID: ' + filter_device_id)
+
+if filter_device_id != '':
+    filter['$and'].append({'id': {'$regex': filter_device_id}})
```

Comment:
It seem you are overriding the `filter` variable in L122. Is this ok? Or I'm missing something?